src/main/java/moze_intel/projecte/emc/NormalizedSimpleStack.java

```diff
@@ -116,17 +123,23 @@ public static NormalizedSimpleStack forOreDictionary(String oreDictionaryName)
 	public static class NSSItem extends NormalizedSimpleStack {
 		public final String itemName;
 		public final int damage;
-		private NSSItem(String itemName, int damage) {
+		public final NBTTagCompound nbt;
+		private NSSItem(String itemName, int damage, NBTTagCompound nbt) {
 			this.itemName = itemName;
 			if (Item.itemRegistry.getObject(itemName) == null) {
 				throw new IllegalArgumentException("Invalid Item with itemName = " + itemName);
 			}
 			this.damage = damage;
+			this.nbt = nbt;
+		}
+		
+		private NSSItem(String itemName, int damage){
+			this(itemName, damage, null);
 		}
 
 		@Override
 		public int hashCode() {
-			return itemName.hashCode() ^ damage;
+			return itemName.hashCode() ^ damage ^ (nbt == null ? 0 : nbt.hashCode());
```

Comment:
Does nbt implement the hash code function properly? If it falls back to the object implementation this will cause problems, because objects hash method uses references.